### DISCUSSION TITLE
search for GitHub repositories using version 4 of the GitHub API (https://developer.github.com/v4)

### DISCUSSION THREAD

**Discussion by GowsikChandran** [community/community#24896] [Apps, API and Webhooks]
<p> search for GitHub repositories using version 4 of the GitHub API (<a href="https://developer.github.com/v4" rel="nofollow noopener">https://developer.github.com/v4</a>)</p>
<p>Please help with some url to test in Postman…</p>
<p>Thanks in advance</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>You can use the <code>search</code> query to search for all kinds of GitHub objects. If I wanted to search for repositories containing “atom” though, I would use:</p>
<pre><code>{
  search(query: "atom", type: REPOSITORY, first: 10) {
    edges {
      node {
        ... on Repository {
          nameWithOwner
        }
      }
    }
  }
}
</code></pre>
<p>which results in:</p>
<pre><code>{
  "data": {
    "search": {
      "edges": [
        {
          "node": {
            "nameWithOwner": "atom/atom"
          }
        },
        {
          "node": {
            "nameWithOwner": "AtomicGameEngine/AtomicGameEngine"
          }
        },
        {
          "node": {
            "nameWithOwner": "rybalkinsd/atom"
          }
        },
        {
          "node": {
            "nameWithOwner": "file-icons/atom"
          }
        },
        {
          "node": {
            "nameWithOwner": "futantan/atom"
          }
        },
        {
          "node": {
            "nameWithOwner": "atomix/atomix"
          }
        },
        {
          "node": {
            "nameWithOwner": "Glavin001/atom-beautify"
          }
        },
        {
          "node": {
            "nameWithOwner": "atom/markdown-preview"
          }
        },
        {
          "node": {
            "nameWithOwner": "amaneureka/AtomOS"
          }
        },
        {
          "node": {
            "nameWithOwner": "atom/apm"
          }
        }
      ]
    }
  }
}
</code></pre>
<p>The <code>... on Repository</code> tells GraphQL that you want to only look at items of type <code>Repository</code>.</p>
